legate/__main__.py

```diff
@@ -14,1112 +14,11 @@
-        help="Specify the size of eager allocation pool in percentage",
-    )
-    parser.add_argument(
-        "--cpu-bind",
-        help="CPU cores to bind each rank to. Comma-separated core IDs as "
-        "well as ranges are accepted, as reported by `numactl`. Binding "
-        "instructions for all ranks should be listed in one string, separated "
-        "by `/`.",
-    )
-    parser.add_argument(
-        "--mem-bind",
-        help="NUMA memories to bind each rank to. Use comma-separated integer "
-        "IDs as reported by `numactl`. Binding instructions for all ranks "
-        "should be listed in one string, separated by `/`.",
-    )
-    parser.add_argument(
-        "--gpu-bind",
-        help="GPUs to bind each rank to. Use comma-separated integer IDs as "
-        "reported by `nvidia-smi`. Binding instructions for all ranks "
-        "should be listed in one string, separated by `/`.",
-    )
-    parser.add_argument(
-        "--nic-bind",
-        help="NICs to bind each rank to. Use comma-separated device names as "
-        "appropriate for the network in use. Binding instructions for "
-        "all ranks should be listed in one string, separated by `/`.",
-    )
-    parser.add_argument(
-        "--launcher-extra",
-        dest="launcher_extra",
-        action="append",
-        default=[],
-        required=False,
-        help="additional argument to pass to the launcher (can appear more "
-        "than once)",
-    )
-    args, opts = parser.parse_known_args()
-    # See if we have at least one script file to run
-    console = True
-    for opt in opts:
-        if ".py" in opt:
-            console = False
-            break
-    if console and not args.not_control_replicable:
-        print("WARNING: Disabling control replication for interactive run")
-        args.not_control_replicable = True
-    ranks = args.nodes * args.ranks_per_node
-    return run_legate(
-        ranks,
-        args.ranks_per_node,
-        args.cpus,
-        args.gpus,
-        args.openmp,
-        args.ompthreads,
-        args.utility,
-        args.sysmem,
-        args.numamem,
-        args.fbmem,
-        args.zcmem,
-        args.regmem,
-        opts,
-        args.profile,
-        args.dataflow,
-        args.event,
-        args.logdir,
-        args.user_logging_levels,
-        args.log_to_file,
-        args.keep_logs,
-        args.gdb,
-        args.cuda_gdb,
-        args.memcheck,
-        args.module,
-        args.nvprof,
-        args.nsys,
-        args.nsys_targets,
-        args.nsys_extra,
-        args.progress,
-        args.freeze_on_error,
-        args.mem_usage,
-        args.not_control_replicable,
-        args.launcher,
-        args.verbose,
-        args.gasnet_trace,
-        args.eager_alloc,
-        args.cpu_bind,
-        args.mem_bind,
-        args.gpu_bind,
-        args.nic_bind,
-        args.launcher_extra,
-    )
-
+from .driver import main
 
 if __name__ == "__main__":
```

Comment:
Is this file still necessary?

Reply:
Not strictly. If we can ever get rid of custom install directories then we could point setuptools here for a normal entry point, rather than maintaining the `legate.py` script. I'd personally rather keep it and keep hope alive ;)

Reply:
I mean that the file `legate/__main__.py` is not used anywhere anymore (except `pyproject.toml`), instead `bin/legate` is using the new `driver/main.py`, so I was suggesting that the old `legate/__main__.py` can be removed.

Reply:
Right that was my understanding, I'd still prefer to keep a `__main__` for now, it affords `python -m legate` 
```
bldtest ❯ python -m legate      
WARNING: Disabling control replication for interactive run
Welcome to Legion Python interactive console
>>> 
```
which I think could potentially be useful with `pytest` later.